core/src/test/java/com/google/cloud/sql/core/CloudSqlInstanceTest.java

```diff
@@ -249,6 +294,77 @@ public void testCloudSqlRefreshesExpiredData() throws Exception {
         () -> instance.getSslData(TEST_TIMEOUT_MS) == data.getSslData(), 1000L);
   }
 
+  @Test
+  public void testThatForceRefreshBalksWhenAScheduledRefreshIsInProgress() throws Exception {
+    // Set expiration 1 minute in the future, so that it will trigger a scheduled refresh
+    // and won't expire during this testcase.
+    InstanceData aboutToExpireData =
+        new InstanceData(
+            null, new SslData(null, null, null), Instant.now().plus(1, ChronoUnit.MINUTES));
```

Comment:
Since we control time here, shall we just make this 5 seconds or something shorter to make our tests fast?

Reply:
I opted for 1 minute because I wanted to trigger a refresh, but I wanted to make sure the initial data does not expire during the test. This test takes less than 1 second to run on my laptop, but can take up to 5 seconds on some of the slower github action executors.